src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,552 @@
+		Long: `'rpk debug bundle' collects environment data that can help debug and diagnose
+issues with a redpanda cluster, a broker, or the machine it's running on. It
+then bundles the collected data into a zip file.
+
+The following are the data sources that are bundled in the compressed file:
+
+ - Kafka data: Number of brokers, consumer groups, topics & partitions, topic
+  configuration and replication factor.
+ - /proc/cpuinfo: CPU information like make, core count, cache, frequency.
+ - /proc/interrupts: IRQ distribution across CPU cores.
+ - redpanda logs: The redpanda logs written to journald. If --logs-since or
+  --logs-until are passed, then only the logs within the resulting time frame
+  will be included.
+ - redpanda configuration: The redpanda configuration file (redpanda.yaml).
+ - Resource usage data: CPU usage percentage, free memory available for the
+  redpanda process.
+ - Clock drift: The ntp drift as reported by 'clockdiff'. Needs CAP_NET_RAW,
+  so it's only included if 'rpk debug bundle' is run as root.
+ - Network config: As reported by 'ifconfig'.
+ - DNS: The DNS info as reported by 'dig', using the hosts in
+  /etc/resolv.conf.
+ - Virtual memory stats: As reported by 'vmstat'.
+ - Socket info: The active sockets data output by 'ss'.
+ - Running process info: As reported by 'top'.
+ - Data directory structure: The 'tree' output describing the data directory
+  contents.
+`,
+		SilenceUsage: true,
+		RunE: func(ccmd *cobra.Command, args []string) error {
+			mgr := config.NewManager(fs)
+			conf, err := mgr.ReadOrFind(configFile)
+			if err != nil {
+				return err
+			}
+			// Since we want to use and bundle in the actual config,
+			// common.FindConfigFile can't be used (it defaults to returning
+			// config.Default()). Therefore, we check that the config is in a
+			// the path given by --config or a default path as defined in
+			// Manager#ReadOrFind.
+			configClosure := func() (*config.Config, error) {
+				return conf, nil
+			}
+			brokersClosure := common.DeduceBrokers(
+				common.CreateDockerClient,
+				configClosure,
+				&brokers,
+			)
+			tlsClosure := common.BuildKafkaTLSConfig(fs, &enableTLS, &certFile, &keyFile, &truststoreFile, configClosure)
+			kAuthClosure := common.KafkaAuthConfig(&user, &password, &mechanism, configClosure)
+			admin := common.CreateAdmin(brokersClosure, configClosure, tlsClosure, kAuthClosure)
+
+			adm, err := admin()
+			if err != nil {
+				log.Error("Couldn't initialize API admin")
+				return err
+			}
+			defer adm.Close()
+
+			return executeSnapshot(fs, conf, adm, logsSince, logsUntil, timeout)
+		},
+	}
+	command.Flags().DurationVar(
+		&timeout,
+		"timeout",
+		10*time.Second,
+		"The maximum amount of time to wait for child commands (ss, ifconfig, "+
+			"dig, etc.) to be executed. "+
+			"The value passed is a sequence of decimal numbers, each with optional "+
```

Comment:
I don't think this line nor the next two are necessary for the help text output (and it makes the help text too long)

Reply:
I think at least the units part is important. Time can be specified in many formats. How about
```go
		"The maximum amount of time to wait for child commands (ss, ifconfig, "+
			"dig, etc.) to be executed. E.g.  '10000ms', '30s', '1m', '0.6h'",
```
?

Reply:
I think it could be very succinct:
```go
how long to wait for child commands to execute (e.g. '100ms', '30s', '1m')
```
your option is good too

Reply:
Fixed.